src/Components/Steps/ZipcodeStep.tsx

```diff
@@ -0,0 +1,179 @@
+import { useContext } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { FormattedMessage } from 'react-intl';
+import { useForm, Controller } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Button, FormControl, FormHelperText, InputLabel, MenuItem, Select, TextField } from '@mui/material';
+import { FormattedMessageType } from '../../Types/Questions';
+import { FormData } from '../../Types/FormData';
+import { Context } from '../Wrapper/Wrapper';
+import PreviousButton from '../PreviousButton/PreviousButton';
+import { updateScreen } from '../../Assets/updateScreen';
+import ErrorMessageWrapper from '../ErrorMessage/ErrorMessageWrapper.tsx';
+import { getQuestion } from '../../Assets/stepDirectory.ts';
+import * as z from 'zod';
+
+interface ZipcodeStepProps {
+  currentStepId: number;
+}
+
+export const ZipcodeStep = ({ currentStepId }: ZipcodeStepProps) => {
+  const { formData, locale, setFormData } = useContext(Context);
+  const matchingQuestion = getQuestion(currentStepId, formData.immutableReferrer);
+  const navigate = useNavigate();
+  const { uuid } = useParams();
+  const { config } = useContext(Context);
+  const { counties_by_zipcode: countiesByZipcode } = config ?? {};
+  const backNavigationFunction = () => navigate(`/${uuid}/step-${currentStepId - 1}`);
+
+  const numberMustBeFiveDigitsLongRegex = /^\d{5}$/;
+
+  const checkCountyIsValid = ({ zipcode, county }) => {
+    const validCounties = countiesByZipcode[zipcode];
+
+    if (validCounties && Object.keys(validCounties).includes(county)) {
+      return true;
+    }
+    return false;
+  };
+
+  const formSchema = z
+    .object({
+      zipcode: z.string().regex(numberMustBeFiveDigitsLongRegex),
+      county: z.string(),
+    })
+    .refine((data) => checkCountyIsValid(data), { message: 'invalid county', path: ['county'] });
```

Comment:
The `invalid county` message in the validation schema won't need to be translated because it's not user-facing. It's used internally within the `zod` schema for validation purposes and helps developers understand which validation rule has failed. The actual error messages shown to users are handled on lines `133` and `164`.